### PR TITLE
web: lazy-load `@sentry/browser`

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context

To make our bundle smaller and reduce the initial loading time of the web application, we can lazy-load heavy dependencies like `@sentry/browser`.

## Changes

- Added script tag to the [app.html](https://github.com/sourcegraph/sourcegraph/blob/main/cmd/frontend/internal/app/ui/app.html) that lazy-loads Sentry SDK.
- Removed `@sentry/browser` imports (apart from TS types) from the `web` package.

## Results

The main bundle is **4.04%** smaller.

Screenshot 2021-11-01 at 15 42 57

## Notes

Ad-blocking or script-blocking extensions may [prevent](https://github.com/easylist/easylist/issues/6963) Sentry SDK from being fetched and initialized correctly. The solution here would be to use a server-side pass-through for event reporting and our CDN for serving the init script. Created an issue to investigate further: https://github.com/sourcegraph/sourcegraph/issues/26981.

Closes https://github.com/sourcegraph/sourcegraph/issues/26840

### CODE REVIEW

File: cmd/frontend/internal/app/ui/app.html [hunk 1]
```diff
@@ -39,6 +39,7 @@
 	'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
 	})(window,document,'script','dataLayer','GTM-TB4NLS7');</script>
 	<!-- End Google Tag Manager -->
+	<script src='https://js.sentry-cdn.com/ae2f74442b154faf90b5ff0f7cd1c618.min.js' crossorigin="anonymous"></script>
```
Comment (umpox) [inline cmd/frontend/internal/app/ui/app.html:42]: I guess we can't `defer` this as it might mean we can't guarantee immediate errors are captured?

**Review by umpox — APPROVED** [review]
👍  👍

File: cmd/frontend/internal/app/ui/app.html (hunk 1 above)
Comment (valerybugakov) [reply cmd/frontend/internal/app/ui/app.html:42]: Yeah, it's a lightweight script that pushes errors to the temporary queue preserved until the SDK is loaded.

File: client/web/src/components/ErrorBoundary.tsx [hunk 2]
```diff
@@ -126,6 +125,11 @@ export class ErrorBoundary extends React.PureComponent<Props, State> {
 }
 
 function shouldErrorBeReported(error: unknown): boolean {
+    // Report errors only in production environment.
+    if (process.env.NODE_ENV !== 'production' && process.env.NODE_ENV !== 'test') {
```
Comment (umpox) [inline client/web/src/components/ErrorBoundary.tsx:129]: How does the mock work with this? It seems that we allow reporting errors in the test env, and the mock actually uses the `sentry/browser` package. Does this mean it actually tries to send Sentry errors (and presumably fails) in our tests?

File: cmd/frontend/internal/app/ui/app.html (hunk 1 above)
Comment (eseliger) [reply cmd/frontend/internal/app/ui/app.html:42]: > Yeah, it's a lightweight script that pushes errors to the temporary queue preserved until the SDK is loaded.

Maybe worth putting in a comment here?

File: client/web/src/components/ErrorBoundary.tsx (hunk 2 above)
Comment (valerybugakov) [reply client/web/src/components/ErrorBoundary.tsx:129]: The `captureException` method is mocked in tests, so it should not report anything.

https://github.com/sourcegraph/sourcegraph/blob/1d5a1187a900e6537411f9378145468e6b5918be/client/web/src/components/ErrorBoundary.test.tsx#L74

**Pull request merged**